GitExtUtils/FileLogging/LogPathGenerator.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.IO;
+
+namespace GitExtUtils.FileLogging
+{
+    public class LogPathGenerator
+    {
+        /// <summary>
+        /// The folder for log files
+        /// </summary>
+        public readonly string LogPath;
+
+        private Tuple<DateTime, string> _fileNameCache = new Tuple<DateTime, string>(DateTime.MinValue, null);
+        private readonly string _logName;
+
+        public LogPathGenerator(string logPath, string logName)
+        {
+            LogPath = logPath;
+            _logName = logName;
+        }
+
+        public string GenerateFullPath(DateTime now)
+        {
+            if (now.Date != _fileNameCache.Item1)
+            {
+                _fileNameCache = Renew(now);
+            }
+
+            return _fileNameCache.Item2;
+        }
+
+        public string GetFilenamePattern()
+        {
+            return $"{_logName}-*.log";
+        }
+
+        private Tuple<DateTime, string> Renew(DateTime now)
+        {
+            var today = now.Date;
+            var fullPath = Path.Combine(LogPath, $"{_logName}-{today:yyyy-MM-dd}.log");
```

Comment:
❗️  This will cause troubles when running concurrently multiple instances of the app

Reply:
What kind of problems are you referring to?

Reply:
Multiple instances attempting to write into the same file.
I saw you use ProcessId in some other place, but I think it should be used here when a file name is generated.